src/nvim/ui_defs.h

```diff
@@ -37,6 +37,8 @@ typedef struct {
   bool ui_ext[kUIExtCount];  ///< Externalized UI capabilities.
   int width;
   int height;
+  int pixel_width;
```

Comment:
If these are changed to floating points, they would be more generally useable for other purposes. For example, to calculate the pixel coordinate of a given cell. Neovim, could internally round this to integers where needed.

Neovide, for example use a fractional grid cell size, so rounding dividing the `pixel_width / width` would give a pretty inaccurate cell width in some cases,

Reply:
I'd be willing to change this, but can you maybe briefly more clearly explain why? Am I getting this right, that neovide uses non-pixel aligned cells in order to fill out the entire window if for example `pixel_width % width != 0`? Couldn't UIs just as well cast the values to float before doing any divisions if needed? Currently, in my mind, the total amount of pixels should always be an integer amount and it should be up to the users of these values to handle division inaccuracies, but I am willing to be convinced otherwise.

Reply:
Let's say my grid width is 10.1 pixels, and the width is 81 columns, so the real with is 818.1. But the UI passes 819.

Now, we want to display something at column 80. Neovim has to calculate the grid width as 819 / 81 = 10.11111, with the information given. Column 80 is 80 * 10.11111 = 808.88889, so you round and display it at pixel 809. While the correct position should be 80 * 10.1 = 808, so it's displayed one pixel wrong. 

It might not matter for the intended use case, but the information could potentially be used for the image protocol for example, and things could become unaligned.

BTW, I'm quite sure that many terminals use fractional grid sizes as well, not just Neovide.

Reply:
Ah, I see, thank you for the clarification. I'll change it, but I am waiting on some clarification in #31903 wrt to what actual types to use.

I'm guessing in terminal buffers we should then always report the floor of the value to prevent applications from overdrawing. Unfortunately `ioctl` only takes shorts for `TIOCSWINSZ`.